File: src/main/java/awktal/mule/Mule.java
package awktal.mule;

public class Mule {

    private MuleType type;

    public Mule() {
        this(MuleType.NONE);
    }

    public Mule(MuleType type) {
        this.type = type;
    }

    public void outfit(MuleType type) {
        this.type = type;
    }

    public MuleType getType() {
        return this.type;
    }
}

File: web/current/web/Mule.dart
import 'MuleType.dart';

class Mule {

    static Map<MuleType, String> imagePaths = {
        MuleType.FOOD: "mule_images/food.png",
        MuleType.ENERGY: "mule_images/energy.png",
        MuleType.ORE: "mule_images/ore.png",
    };

    MuleType type = MuleType.NONE;

    Mule();

    void outfit(t) {
        type = t;
    }

    static String getImagePath(MuleType type) {
        return imagePaths[type];
    }
}